lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2724,6 +2779,9 @@ void ModuleEmitter::prepareHWModule(Block &block) {
     if (auto instance = dyn_cast<InstanceOp>(op)) {
       // Anchor return values to wires early
       lowerInstanceResults(instance);
+      // Anchor ports of bound instances
+      if (instance->hasAttr("doNotPrint"))
+        lowerBoundInstance(instance);
       addLegalName(ValueOrOp(&op), instance.instanceName(), &op);
```

Comment:
Does a bound instance op need to have its name checked for legality?

Reply:
I think the instance name is already legalize earlier in the `HWLegalizeNames` pass, so it should be fine at this point.

Reply:
Yes, the name winds up in the verilog via the bind op output.

Reply:
And at this point, it's adding previously legalized names so anything else doesn't rename already legalized things.